### PR TITLE
Refactor and remove service info dependency to obtain the implementation's drs version

### PR DESCRIPTION
**Pull request by yash-puligundla**
- De-duplicated the code
    - Each DRS endpoint corresponds to a Phase object
    - Each Test object corresponds to a single API call. The response will be tested against multiple test cases and each of these test cases corresponds to a Case object. 
    - Move common test cases into a method. The common test cases include validating status_code, content-type, and schema of the endpoint response. This method will be called in each of the Test objects.
- If a schema_file is missing, throw an error
- If case status = SKIP, skip the case without any other checks
- Add a method for each Test  - test_service_info, test_drs_object_info, test_drs_object_access
- Remove the dependency on service-info to obtain drs version. Instead this information will be obtained from command line arguments
- report_runner returns a Report object instead of JSON

### CODE REVIEW

**Comment by yash-puligundla** [comment]
report_runner returns a Report object instead of JSON

File: compliance_suite/validate_response.py [hunk 1]
```diff
@@ -26,54 +26,58 @@ def set_case(self, case):
...
-                validate(instance = self.actual_response.json(),resolver = reference_resolver,schema=expected_schema)
-                self.case.set_message("Schema Validation Successful")
-                self.case.set_status_pass()
-            except Exception as e:
-                self.case.set_message(e.message if hasattr(e,"message") else str(e))
-                self.case.add_log_message("Stack Trace: "+str(e))
-                self.case.set_status_fail()
-        else:
-            pass
-            # make sure if skip, the reason is added to message and/ log_message
```
Comment (Chen2x) [inline compliance_suite/validate_response.py:51]: I don't think the log was implemented. Should we keep the comment for future?

**Review by Chen2x — APPROVED** [review]
Everything looked pretty good. I think a lot of improvements have been made in terms of reducing repeated code. I noticed that some of the earlier commenting was replaced by well named functions but I think some extra comments might be good for clarity. Especially since some of the new logic you used such as the string formatting and the inline if statements reduce readability slightly. I will take one last pass tomorrow to see if I catch anything.

**Comment by Chen2x** [comment]
Updated Path and everything looks good now

File: compliance_suite/validate_response.py (hunk 1 above)
Comment (yash-puligundla) [reply compliance_suite/validate_response.py:51]: There are no logs implemented. However we do have a logging class, which makes it easier to implement logs. I have added a ticket for logging - [GA4GHTT90](https://www.ebi.ac.uk/panda/jira/browse/GA4GHTT-90)

**Comment by yash-puligundla** [comment]
Thanks for the review, Chen. I am merging the PR.

**Pull request merged**